salt/states/git.py

```diff
@@ -1588,6 +1588,455 @@ def present(name,
     return ret
 
 
+def detached(name,
+           ref,
+           target=None,
+           remote='origin',
+           user=None,
+           force_clone=False,
+           force_checkout=False,
+           fetch_remote=True,
+           hard_reset=False,
+           submodules=False,
+           identity=None,
+           https_user=None,
+           https_pass=None,
+           onlyif=False,
+           unless=False,
+           **kwargs):
+    '''
```

Comment:
Please add a line here at the start of the docstring saying  ``..versionadded:: Boron``, followed by a blank line.